packages/babel-plugin-transform-es2015-parameters/src/default.js

```diff
@@ -44,17 +50,33 @@ const iifeVisitor = {
 };
 
 export const visitor = {
-  Function(path) {
+  Function(path, state) {
     const { node, scope } = path;
     if (!hasDefaults(node)) return;
 
     // ensure it's a block, useful for arrow functions
     path.ensureBlock();
 
-    const state = {
-      iife: false,
-      scope: scope,
-    };
+    const params = path.get("params");
+
+    if (state.opts.loose) {
+      const body = [];
+      for (let i = 0; i < params.length; ++i) {
+        const param = params[i];
+        if (param.isAssignmentPattern()) {
+          body.push(buildLooseDefaultParam({
+            VARIABLE_NAME: param.get("left"),
```

Comment:
I am sorry about this error, but the actual source code is the way you suggested. The template I defined for this is the following:

```javascript
const buildLooseDestructuredDefaultParam = template(`
  let ASSIGMENT_IDENTIFIER = PARAMETER_NAME === UNDEFINED ? DEFAULT_VALUE : PARAMETER_NAME;
`);
```
this will be passed the corresponding values resulting in your version of the example above.